Library/Homebrew/requirements.rb

```diff
@@ -125,16 +125,28 @@ class JavaDependency < Requirement
 
   satisfy { java_version }
 
+  env do
+    java_home = Pathname.new(@java_home)
+    ENV["JAVA_HOME"] = java_home
+    ENV.prepend_path "PATH", java_home/"bin"
+    if (java_home/"include").exist? # Oracle JVM
+      ENV.append_to_cflags "-I#{java_home}/include"
+      ENV.append_to_cflags "-I#{java_home}/include/darwin"
+    else # Apple JVM
+      ENV.append_to_cflags "-I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/"
+    end
+  end
+
   def initialize(tags)
-    # ignore trailing +
-    @version = tags.shift.sub(/\+$/, "") if /(\d\.)+\d/ === tags.first
+    @version = tags.shift if /(\d\.)+\d/ === tags.first
```

Comment:
So the arguments are not passed to the Requirement superclass; it's the way most other Requirements seem to behave.